SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+	// Unfortunately, the nature of some tests requires them to have actors placed in the level, to trigger some Unreal behavior
+	GEditor->AddActor(CurrentLevel, ADormancyTestActor::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AReplicatedVisibilityTestActor::StaticClass(), FTransform::Identity);
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir()
+														   + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast/SpatialNetworkingMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// This map is a simple 2-server-worker map, where both workers see everything in the world
+void UGenerateTestMapsCommandlet::CreateSpatial2WorkerMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the tests
+
+	// The two orchestration tests are placed at opposite points around the origin to "guarantee" they will land on different workers so
+	// that they can demonstrate they work in all situations
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(-250, 250, 0)));
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(250, -250, 0)));
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, APredictedGameplayCuesTest::StaticClass(), FTransform::Identity);
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, ACrossServerAbilityActivationTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart1::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart2::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARelevancyTest::StaticClass(), FTransform::Identity);
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest1x2WorkerSettings::StaticClass());
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(
+		FPaths::ProjectContentDir() + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast_Spatial_Only/Spatial2WorkerMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// This map is custom-made for the SpatialAuthorityTest - it utilizes a gamemode override
+void UGenerateTestMapsCommandlet::CreateSpatialAuthorityMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the test
+	ASpatialAuthorityTest* AuthTestActor = CastChecked<ASpatialAuthorityTest>(
+		GEditor->AddActor(CurrentLevel, ASpatialAuthorityTest::StaticClass(), FTransform(FVector(-250, -250, 0))));
+
+	// Add the helpers, as we need things placed in the level
+	// PRComment: Hmmmmmmmm, these CastChecked with the static class defined right there are kind of annoying
```

Comment:
PRComment ^

Reply:
Don't see a way to change this other than to create a helper function that does this for you - `template <class T> T* AddLevelActor(ULevel*) { return CastChecked<T>(GEditor->AddActor(Level, T::StaticClass())); }`

Reply:
Yeah, that was my idea as well. I think I can do it, just to clean up the code a bit.